src/components/pages/SettingsPage.jsx

```diff
@@ -29,34 +29,96 @@ export default class SettingsPage extends SimplePageBase {
         this.state = {
             oldPassword: '',
             newPassword: '',
+            email: this.props.user.getIn(['data', 'email']),
         };
+
     }
 
     renderContent() {
         const passwordPending = this.props.passwordStore.get('isChangePending');
         const passwordChanged = this.props.passwordStore.get('changed');
         const passwordError = this.props.passwordStore.get('changeError');
-        let msg = null;
+        let passwordMsg = null;
 
         if (passwordChanged) {
             if (passwordError) {
-                msg = <p className="SettingsPage-msg-error" id="pages.settings.password.error">Wrong password</p>
+                passwordMsg = <p className="SettingsPage-msg-error" id="pages.settings.password.error">Wrong password</p>
             }
             else {
-                msg = <p className="SettingsPage-msg-success" id="pages.settings.password.success">Password changed</p>
+                passwordMsg = <p className="SettingsPage-msg-success" id="pages.settings.password.success">Password changed</p>
                 setTimeout( () => {
-                    this.props.dispatch(resetPasswordChanged());
+                    this.props.dispatch(resetEmailChanged());
                 }, 10000);
             }
         }
 
-        let submitLabel = this.props.intl.formatMessage(
+        const emailPending = this.props.user.get('isChangePending');
+        const emailChanged = this.props.user.get('changed');
+        const emailError = this.props.user.get('changeError');
+
+        let emailMsg = null;
+
+        if (emailChanged) {
+            if (passwordError) {
```

Comment:
Is this a mistake? Or is the password error relevant when changing e-mail?

I tried changing to an already used e-mail address and got 409, but no error message was displayed, probably because of this?

Reply:
This was a mistake. The error handling has been revamped  now.